src/Microsoft.AzureHealth.DataServices.Core/Microsoft.AzureHealth.DataServices.Core.csproj

```diff
@@ -20,9 +20,9 @@
     <PackageReference Include="Microsoft.Extensions.Logging.ApplicationInsights" Version="2.21.0" />
     <PackageReference Include="Microsoft.IdentityModel.JsonWebTokens" Version="6.23.1" />
     <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
-  </ItemGroup>
+  </ItemGroup>  
 
   <ItemGroup>
-    
+    <ProjectReference Include="..\Microsoft.AzureHealth.DataServices.Caching\Microsoft.AzureHealth.DataServices.Caching.csproj" />
```

Comment:
I don't think using the caching that is part of this toolkit is the best way to implement this. The caching in the toolkit is meant to be used by the consumers of the toolkit. We will be relying on users to add caching to enable the token cache, which makes the toolkit harder to use.

I'd investigate adding an internal cache type (`AccessTokenCache` or something like that) that the user does not need to setup manually. For every call of the HTTP Client for that type of the binding, the client would use the cached value if it exists. If the token is expired, it would attempt to get the token again. 

The Azure SDKs (see links in PR) accomplishes this via a Policy for their clients. These are simpler caching strategies online / NuGet packages. This also can wait if there is concerns around proper implementation.

Reply:
Hi Mikael,

  We removed the previous implementation and added the token cache logic within Microsoft.AzureHealth.DataServices.Core/Security so that the user does not need to configure anything manually. Instead, it will cache the token on the first call and read it from cache on subsequent calls.
 
We have also added the IHttpClientFactory in RestRequest for HttpClient object creation, as suggested by Matt.
[Poor Performance using Authenticator and RestBinding](https://github.com/microsoft/azure-health-data-services-toolkit/issues/92)

Can you please take a look at the updated code and let us know if you require any changes?